src/error.js

```diff
@@ -1,11 +1,22 @@
 /* Error handling.
  */
-import { get } from 'lodash';
 import {
+    GATEWAY_TIMEOUT,
     INTERNAL_SERVER_ERROR,
     NOT_FOUND,
-    GATEWAY_TIMEOUT,
 } from 'http-status-codes';
+import { get, omit } from 'lodash';
+
+// We are accidentally leaking auth headers to loggly when throwing
+// an OpenAPIError. We want to fix this properly in the long term,
+// but as a short term solution we are simply going to omit the headers
+// from the error here. This operation is case insensitive.
+function pruneAuthorizationHeaders(headers) {
+    if (!headers || !Object.keys(headers).map(key => key.toLowerCase()).includes('authorization')) {
+        return headers;
+    }
+    return omit(headers, ['authorization', 'Authorization']);
```

Comment:
I don't particularly like that the `if` statement and the `omit` line don't match: one is case-insensitive, the other only deals with a case-insensitive `A`. Cloning objects can be expensive but something like:

```suggestion
function pruneAuthorizationHeaders(headers) {
    return pruneObject(headers, ['authorization']);
}
function pruneObject(obj, disallowed=[]) {
    const clone = shallowClone(headers);
    Object.keys(clone).forEach(key=>{
        if (disallowed.includes(key.toLowerCase().trim())) delete clone[key];
    });
    return clone;
```

Feels like a more consistent approach. This behavior might be doable with `_.omit` or some other function (`_.omitBy`? I swear at least one of them takes a predicate function) to get those various hyperoptimizations. The only reason it needs a clone at all is so we don't delete that header from the source object, don't know if we can determine it's safe to modify it at this point (is only in an error constructor so we might be able to say definitively that anything using that data is completed)

Reply:
omit() returns a new object, it doesn't mutate the original object

Reply:
Yeah, `omit` does a shallow clone too, I more referring to my suggestion that always makes a copy (as opposed to the original that may make a copy).

Finally did look at lodash docs, `omitBy` (or `pluckBy`) does take a predicate, what about:

```suggestion
function pruneAuthorizationHeaders(headers) {
    const disallowed = ['authorization'];
    return _.omitBy(headers, (val,key)=>{
        return disallowed.includes(key.toLowerCase().trim());
    });
```
?